code/espurna/config/general.h

```diff
@@ -621,19 +621,74 @@
 #define SPIFFS_SUPPORT              0           // Do not add support for SPIFFS by default
 #endif
 
+// -----------------------------------------------------------------------------
+// SSL Client                                                 ** EXPERIMENTAL **
+// -----------------------------------------------------------------------------
+
+#ifndef SECURE_CLIENT
+#define SECURE_CLIENT                          SECURE_CLIENT_NONE     // What variant of WiFiClient to use
+                                                                      // SECURE_CLIENT_NONE    - No secure client support (default)
+                                                                      // SECURE_CLIENT_AXTLS   - axTLS client secure support (All Core versions, ONLY TLS 1.1)
+                                                                      // SECURE_CLIENT_BEARSSL - BearSSL client secure support (starting with 2.5.0, TLS 1.2)
+                                                                      //
+                                                                      // axTLS marked for derecation since 2.4.2 and **will** be removed in the future
+#endif
+
+// Support Maximum Fragment Length Negotiation TLS extension
+// "...negotiate a smaller maximum fragment length due to memory limitations or bandwidth limitations."
+// - https://arduino-esp8266.readthedocs.io/en/latest/esp8266wifi/bearssl-client-secure-class.html#mfln-or-maximum-fragment-length-negotiation-saving-ram
+// - https://tools.ietf.org/html/rfc6066#section-4
+#ifndef SECURE_CLIENT_MFLN
```

Comment:
Just one thing I thought about: this should mention that it's only working with ``SECURE_CLIENT`` set to ``SECURE_CLIENT_BEARSSL``

Reply:
Also, the value should be between 512 and 16384:
https://github.com/esp8266/Arduino/blob/c18b402c31dc27ffc4b8817035ad8a44670c1b2e/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp#L186-L187

Reply:
Come to think of it, recv and send values can be different. Default sets it as high as possible for receiving end (since server can just dump a lot of data and expect it to be parsed):
https://github.com/esp8266/Arduino/blob/c18b402c31dc27ffc4b8817035ad8a44670c1b2e/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp#L83

But, MFLN *only* supported values are 512, 1024, 2048 and 4096. 16k is maximum record size for TLS in general.
https://tools.ietf.org/html/rfc6066#page-8
https://tools.ietf.org/html/rfc8449#page-3

Reply:
Ah, good to know - I see the same is documented here:
https://github.com/esp8266/Arduino/blob/c18b402c31dc27ffc4b8817035ad8a44670c1b2e/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp#L1359-L1361

Reply:
And down below the exact protocol byte sequences:
https://github.com/esp8266/Arduino/blob/c18b402c31dc27ffc4b8817035ad8a44670c1b2e/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp#L1399-L1405

It is actually mentioned in the documentation proper! Missed that (since I looked as setBufferSizes)
> probeMaxFragmentLenght() ... (len must be a power of two from 512 to 4096, per the specification)